src/vs/editor/browser/services/openerService.ts

```diff
@@ -51,12 +56,42 @@ export class OpenerService implements IOpenerService {
 
 	private _doOpen(resource: URI, options?: { openToSide?: boolean }): Promise<boolean> {
 
-		const { scheme, path, query, fragment } = resource;
+		const { scheme, authority, path, query, fragment } = resource;
 
-		if (equalsIgnoreCase(scheme, Schemas.http) || equalsIgnoreCase(scheme, Schemas.https) || equalsIgnoreCase(scheme, Schemas.mailto)) {
-			// open http or default mail application
+		if (equalsIgnoreCase(scheme, Schemas.mailto)) {
+			// open default mail application
 			return this.openExternal(resource);
+		}
 
+		if (equalsIgnoreCase(scheme, Schemas.http) || equalsIgnoreCase(scheme, Schemas.https)) {
+			const trustedDomains = this._configurationService.getValue<string[]>('http.trustedDomains');
+			const domainToOpen = `${scheme}://${authority}`;
+
+			if (isDomainTrusted(domainToOpen, trustedDomains)) {
+				return this.openExternal(resource);
+			} else {
+				return this._dialogService.confirm({
+					title: localize('openExternalLink', 'Open External Link'),
+					type: 'question',
+					message: localize('openExternalLinkAt', 'Do you want to leave VS Code to open the external website at') + ` ${resource.toString()}?`,
+					detail: resource.toString(),
```

Comment:
I think "leave" isn't a good word here and I would remove the `to leave` part.

Reply:
Also, the string isn't correctly externalised, the `{0}` syntax must be used

Reply:
Last, the detail seems to repeat what the message says? It can also get v e r y long with path, query, and fragment and also not that `toString` encodes - pass `true` for a human readable version

Reply:
All good points. Taken.